### ISSUE TITLE
Building Footprint fixes based on OSE list (SOW task 2.1)

### ISSUE THREAD

**Issue by danrademacher** [GreenInfo-Network/seattle-building-dashboard#99]
> Task 2.1 Prescribed Footprint Corrections
> Description: Informed by OSE documentation, correct instances where property footprints are currently inaccurate. (e.g. multi-building properties represented by a single shape, shape is located in the wrong spot, shape is incomplete) There are an estimated 250 property footprints that require modification.

Mike shared a 200MB(!) Excel file. It is so large that Excel online will not load it, and Sharepoint doesn't have an easy method, on Mac anyway, to open a shared file with desktop Excel in a way that doesn't cause version control issues.

I copied the file here:
https://docs.google.com/spreadsheets/d/1Xr2P3XwoaEl61QEHZAsdoTY9eu5DJZ-J?rtpof=true&usp=drive_fs

There are already 375 issues logged here -- so 75 more than in the scope.

I wanted to see them all on a map so I ran this in CARTO using a filtered building list:

```sql
SELECT * FROM public.seattle_building_outlines_2022 where buildingid in (41,43,73,84,87,114,125,144,168,172,175,185,190,194,198,205,212,215,217,225,226,227,234,245,247,248,249,266,268,281,293,302,315,325,326,345,350,351,355,358,364,381,422,435,467,477,496,500,516,576,608,658,683,689,690,694,708,728,730,819,1281,19489,19510,19584,19694,19696,19704,19754,19777,19780,19788,19831,19847,19848,19849,19873,19925,19947,19990,19992,20027,20028,20045,20047,20052,20056,20082,20095,20100,20140,20187,20216,20217,20234,20241,20242,20321,20340,20367,20369,20456,20490,20522,20523,20526,20543,20585,20659,20686,20720,20762,20805,20810,20888,20930,21117,21122,21150,21196,21202,21215,21216,21218,21231,21310,21314,21335,21341,21382,21399,21401,21425,21432,21476,21533,21556,21557,21570,21604,21610,21616,21627,21692,21697,21745,21811,21941,22259,22261,22264,22299,22334,22504,22511,22607,22654,22739,22742,22801,22810,22863,22939,22941,22954,22964,23020,23028,23050,23074,23076,23173,23242,23265,23322,23323,23327,23400,23409,23484,23495,23496,23503,23554,23605,23615,23622,23632,23637,23639,23714,23730,23752,23846,23937,23956,24030,24036,24086,24150,24160,24161,24162,24179,24235,24248,24253,24306,24489,24494,24495,24496,24560,24562,24587,24599,24617,24645,24659,24750,24775,24854,24857,24934,24987,25053,25054,25129,25131,25241,25245,25251,25299,25360,25371,25410,25435,25527,25543,25652,25710,25761,25973,26007,26026,26040,26098,26150,26198,26283,26313,26450,26460,26558,26574,26729,26746,26748,26808,26821,26862,26880,27008,27071,27231,27241,27265,27300,27342,27343,27348,27389,27597,27600,27774,27799,27833,27888,27901,27903,27955,27970,27981,28006,28020,28025,28874,29069,29170,29310,29312,29550,29612,29792,30207,31187,32454,33527,37008,37248,39332,40388,40447,41068,41070,45287,48287,49699,49703,49704,49705,49709,49721,49722,49731,49735,49736,49737,49738,49771,49792,49797,49821,49825,49862,49889,49894,49911,49943,49945,49946,49967,49990,49996,49998,50002,50003,50005,50009,50023,50038,50049,50055,50067,50070,50101,50160,50175,50177,50180,50194,50245,50247,50259,50294,50309,50347,50376,50383,50385,50415,50416,50462,50481,50524,50642,50656,50693,50694,50704,50705,50710,50718,20066,103,400)
```

Here's a map of those:
https://cityenergy-seattle.carto.com/builder/555fefed-6afe-4777-9d40-25de320e2332/embed

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201108652060192/1207476636390786)
┆Due Date: 2024-06-14

**Comment by danrademacher** [comment]
A bit surprised to see so many UW main campus buildings in there!

Mike says:
>UW Campus record has many little weird shape nubs. And buildings aren't all 100% correct. There are some overlapping shapes and things that aren't actually buildings displaying. Compare to the UW Shapefile to remove miscellaneous nubs and correct shapes. -MR

The weird thing is that I am pretty sure we took these buildings from the UW shapefile!

**Comment by danrademacher** [comment]
See this old ticket where we did pull those from UW shp, though there were a lot of oddities and edge cases:
https://github.com/GreenInfo-Network/seattle-building-dashboard/issues/86

**Comment by danrademacher** [comment]
The Images tab has _very detailed_ hand drawn screenshots of what Mike wants to see. Wow, this work is going to be tedious. He really needs to learn GIS so instead of hand drawing this stuff and then giving it to us, he can just fix it! I think it might be about the same amount of work for him.

The Images tab has 178 building IDs. So not sure yet what's up there with 375 noted for edits in the main tab and 178 documented visually.

**Comment by danrademacher** [comment]
I spoke to Mike Roos and determined that the 178 building IDs in his screenshots tab are the ones we need to fix. The other tab also includes items we have fixed in the past.

Among the challenges are that his screenshot tab is impossible to open in Sheets and hard to track and manage generally, especially relative to the process outlined at https://github.com/GreenInfo-Network/seattle-building-dashboard/blob/master/documentation/Building_outlines.md

Probably the best thing would be to make a new, concise list of the IDs he wants to change, then characterize the changes as Replacements, Additions, or Deletions. 

But given the detail of his instructions, we might need to just digitize all these changes rather than trying to grab things from the open data footprints (`P:\proj_p_s\Seattle Building Dashboard\2023 update\Building_Outline_2015`). Though one could load the 2015 data as reference and pull in relevant footprints when they better align with Mike's requests. My hunch though is we're no in places where the old outlines are just incorrect and we need to hand edit the data to better match reality.

Brought down latest footprint from CARTO data here: P:\proj_p_s\Seattle Building Dashboard\2024_update

**Comment by joseph-stout** [comment]
Shapefile with updated building footprints, using aerial imagery and OSM basemap is now here: 
P:\proj_p_s\Seattle Building Dashboard\2024_update\buidlingoutlines_updates_20240628

**Comment by danrademacher** [comment]
Reviewing changes:

## Building ID 49736
Based on Mike's notations in Excel, I think we need to add these other buildings to 49736:
![image](https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/c5e6a6a8-ca70-4f69-8978-e6236b746cb0)

His notation:
<img width="1139" alt="image" src="https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/9b8b43df-2038-467f-8196-e2fad19df819">

## 20543
Missing this change:
![image](https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/b7f16bb2-17a5-4c42-a5c5-58a28571dd12)

Based on mike's notes, we need to add 2 missing buildings to the outlines for this record (upper right and lower left):
<img width="1199" alt="image" src="https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/f6e27ff1-b5e5-4525-be4b-e2be8404f243">


## 50070
Not sure if Mike Roos meant us to only add a second building here or also remove the existing building
![image](https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/2ace117c-4d67-4c5d-8cec-7b93620016a6)

He usually draws an X through things to remove so my hunch is this was an add, but it's hard to tell. If there's no other evidence you happened upon, we can just ask him
<img width="1302" alt="image" src="https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/dedcc641-e1ab-4cb7-b508-421727e130a9">

## 20045
Hmm, where is this building?
![image](https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/2c19733d-9e2e-4e5d-9811-fb9cd635c5b5)

This is Mike's note:
<img width="1193" alt="image" src="https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/5afe3626-86e5-44cc-b32e-90155f848856">

I t looks like the ID got deleted from the geometry that should be 20045 but is 0:
![image](https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/1b2b04b0-c092-40b0-a956-08e98602953c)

## 20047
This one is also unclear. It should be somewhere here:
![image](https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/f8b6bfef-38a9-4576-af1b-b31b9ccad79d)

But I don't understand Mike's instructions at all
<img width="1361" alt="image" src="https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/21e829ea-f199-48d6-8f63-ce66158f3cb2">

What change did you make here?

**Comment by danrademacher** [comment]
## 27888

Did this one change at all?

It looks like Mike wants to simplify the outline and make it look like an L without the little rings on the south side:
![image](https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/2fdc84b0-5df4-40d8-82ad-ad4012f5dcd5)

<img width="966" alt="image" src="https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/fef668ff-3b3b-4f8c-9382-c76a107681b1">

## 22264 and 22261
This one still has an overlap in it that needs to be fixed. Looks like the 22261 on the left can be deleted:
![image](https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/770d71a2-a8b7-407a-86b9-3a2c0d604c5b)

Mike's sketch
<img width="799" alt="image" src="https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/58271da3-bf32-4da3-b185-c34f1fa4fcaa">

## 22299, 49946
Still a little overlap that could be fixed
![image](https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/8e58f102-d3b4-4a53-8a72-50218254c925)

## 24987
There's a tiny sliver of this building hanging around that should be deleted
![image](https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/edd82f31-e030-4bee-bb92-2282ea0068db)

## 24934
Missed adding a footprint here:
![image](https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/abbd9bab-e444-4da0-906f-c2c42dfa815d)

There should be 3:
<img width="1347" alt="image" src="https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/8f44b3c5-b068-408b-98db-bb1c0ac0e190">

## 23622
Looks like we missed one or two small buildings at the far right:
![image](https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/e0bcf42d-aee9-447d-90e2-80aae691139a)

<img width="1299" alt="image" src="https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/e1469898-3f6f-4670-9e4d-5b26242197db">

## 26283
This should have 3 footprints:
![image](https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/050efb8f-15b0-4297-9932-73a1fb959acc)

<img width="1096" alt="image" src="https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/2d6ea324-220d-4583-bb0b-9a73ebe522e8">

## 19788
The lower left outline should be 19788. Mike has marked both these outlines as 19788 but we have one as 49730:
![image](https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/1612b559-0232-48a5-8447-10f8e476cd3b)

<img width="1044" alt="image" src="https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/453cf550-1d08-4390-96d9-7b782b17f5e0">

## 27071
This shape got broken during edits:
![image](https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/9a9f082a-9fab-47c7-9672-227f09df0994)

<img width="1149" alt="image" src="https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/9f87f856-d155-40a7-b0ed-aa27b7144452">

## 28006
The right footprints are here, but 2 of them have null buildingid and should be 28006
![image](https://github.com/GreenInfo-Network/seattle-building-dashboard/assets/1423200/a023aac0-f6e1-4859-8b55-48df5428c051)

that's as far as I got on specific footprint QA, thorugh ID 28006 (Row 2437) of the sheet.

@joseph-stout can you go through these and then double check the work in the rest of the sheet one by one to confirm changes?

One thing I just noticed is that there are 14 footprints with `buildingid=0` in at least three cases, those were changes you made but somehow the IDs got deleted. I suspect the others might be similar. Worth checking those.

**Comment by joseph-stout** [comment]
Yes sorry. I will go back through, I didn't realize it was so rough. There were may cases where I found his instructions unclear, but the missing building ID's etc. are clearly errors on my part.

**Comment by joseph-stout** [comment]
I picked up from 28006 on and reviewed the rest of the building edits. I made a few more corrections, and merged the final two building footprints that Tom digitized and uploaded. This should now be complete.

**Comment by joseph-stout** [comment]
P:\proj_p_s\Seattle Building Dashboard\2024_update\seattle_building_outlines_2024_update\seattle_building_outlines_2024_update

**Comment by danrademacher** [comment]
These are showing lockfiles so I can't ZIP these up:
<img width="520" alt="image" src="https://github.com/user-attachments/assets/0fef8914-1f16-4939-8f0f-fa8a9280d4f8">

On wed, can you make sure they're not still open on your machine?

**Comment by joseph-stout** [comment]
Sorry about that, I had forgotten to close the project on my machine. But it is closed now.

**Comment by danrademacher** [comment]
Great. This loks good. Tom was able to get the file and deploy the update to our staging environment.

**Comment by joseph-stout** [comment]
Re-opened this ticket to share the project destination that contains the digitized footprints, and some workflow notes.

Project path:
P:\proj_p_s\Seattle Building Dashboard\2023 update\DigitizingFootprints_20230921\Digitzing_Footprints.aprx

Notes:
I have found referencing the aerial imagery first to confirm correct location, then using the OSM base map to trace the footprint works best. There may be some cases where a judgment call is required as the imagery may capture portions of the building that are not included in OSM. 

I also found the ESRI basemap did not have a fine enough resolution for individual buildings to be easily visible, so I added Google imagery as a tiled service layer which is much clearer. An added benefit of having the imagery overtop of the basemap vs. making it the basemap itself is it is easy to toggle back and forth. Previously I was cross-referencing with Google Maps from lat./long. but obviously this was less efficient. 

Some Edit tools in ArcGIS Pro that are useful:
_Create feature_-- to add a new feature
_Edit vertices_-- to adjust or reshape (can add/remove vertices by right-clicking on a vertex or line)
_Move_ -- in cases where the building shape may be offset somewhat from the correct footprint location.
_Continue feature_-- to trace "holes"  in a building area such as courtyards. This allows cutouts within the feature without creating weird lines or fragments in the footprint polygon that would otherwise occur.
_Merge_ -- to make multiple polygons into one feature. It is important to ensure the correct attributes are retained
